### PR TITLE
fix: Prevent US paid Youtube player reinitialising itself when trying to mute

### PR DESCRIPTION
**Pull request by chrislomaxjones**
Co-authored-by: Dominik Lander <dominik.lander@guardian.co.uk>

## What does this change?

Fix a bug on Paid for US pages when using Android, where the player repeatedly reinitialises itself.

Use the mute function available on the Youtube Player, instead of muting by updating the `src` attribute on the Youtube player iframe. This removes the need for the `muteIFrame` function, so we remove its tests as well.

## Does this change need to be reproduced in dotcom-rendering ?

- [X] No
- [ ] Yes (please indicate your plans for DCR Implementation)

## What is the value of this and can you measure success?

This fixes a bug on Android devices where the Youtube Player repeatedly flashes between the player and the overlay.

The suspicion was that updating the iframe `src` attribute causes the player to reinitialise itself. This causes the iframe `src` to be repeatedly updated in a loop, causing the flashing. Removing this logic and using the `event.target.mute()` should alleviate this issue. 

### Tested

- [X] Locally
- [x] On CODE (optional)

### CODE REVIEW

**Review by arelra — APPROVED** [review]
Nice fix!

If you haven't already it might be worth verifying we didn't introduce any regressions on pages where Frontend's player is still used:

Video front: https://www.theguardian.com/video
Video page: https://www.theguardian.com/uk-news/video/2022/may/11/made-in-liverpool-this-land-is-our-land-video

File: static/src/javascripts/projects/common/modules/atoms/youtube.js [hunk 1]
```diff
@@ -373,7 +365,7 @@ const onPlayerReady = (
     const iFrameBehaviourConfig = getIFrameBehaviourConfig(iframe);
     const iFrameBehaviour = getIFrameBehaviour(iFrameBehaviourConfig);
     if (iFrameBehaviour.mutedOnStart) {
-        muteIFrame(iframe);
+        event.target.mute();
```
Comment (SiAdcock) [inline static/src/javascripts/projects/common/modules/atoms/youtube.js:368]: I think `event.target` gets assigned to `youtubePlayer` [a bit further up](https://github.com/guardian/frontend/pull/25199/files#diff-ef828fcf95131da1b9fcb893ba39b1e82b00da5dd2deecda295303148ff4cc95R350)
```suggestion
        youtubePlayer.mute();
```

**Review by SiAdcock — APPROVED** [review]
Lovely. Nice work simplifying the `isUsPaidContentVideo` bit too 👌

**Review by zekehuntergreen — APPROVED** [review]
Looks good!

File: static/src/javascripts/projects/common/modules/atoms/youtube.js (hunk 1 above)
Comment (chrislomaxjones) [reply static/src/javascripts/projects/common/modules/atoms/youtube.js]: Nice spot, have updated the PR with this.

**Comment by chrislomaxjones** [comment]
> If you haven't already it might be worth verifying we didn't introduce any regressions on pages where Frontend's player is still used

Good shout, I just did a check and all looks okay to me. I think we're lucky this change is gated by `isUsPaidContentVideo`.

**Pull request merged**